docs/roles.md

```diff
@@ -0,0 +1,104 @@
+# Roles
+
+## Computation Roles
+
+Computation roles are used to manage user actions to a computation. On service start, builtin roles are created with their associated actions which can be assigned to a user on a particular computation. These builtin roles can be updated, deleted, and new ones added.
```

Comment:
more details on computation roles and what purpose they serve on the manifest. also what each allows you to do. Also we need to emphasize that a user can only belong to a single role and if custom actions are required for example giving a user algo provider and dataset provider role at the same time a new role with those actions need to be created